test/parallel/test-http-writable-true-after-close.js

```diff
@@ -7,29 +7,40 @@ const { get, createServer } = require('http');
 // res.writable should not be set to false after it has finished sending
 // Ref: https://github.com/nodejs/node/issues/15029
 
+let internal;
 let external;
 
-// Http server
-const internal = createServer((req, res) => {
-  res.writeHead(200);
-  setImmediate(common.mustCall(() => {
-    external.abort();
-    res.end('Hello World\n');
-  }));
-}).listen(0);
-
 // Proxy server
 const server = createServer(common.mustCall((req, res) => {
   get(`http://127.0.0.1:${internal.address().port}`, common.mustCall((inner) => {
-    res.on('close', common.mustCall(() => {
+    const listener = common.mustCall(() => {
```

Comment:
Is this called more than once on any platforms? Removing the listeners kind of defeats the purpose of wrapping it in `common.mustCall()`.

Reply:
I guess the point is to ensure that the listener is actually called.

Reply:
Yes, but `common.mustCall()` also enforces the number of times the function is called. Removing the listeners means we lose that checking.

Reply:
I think either `'close'` or `'finish'` should be emitted so it doesn't matter as long as one at least one is emitted.

Reply:
That's why I asked:

> Is this called more than once on any platforms?

We shouldn't remove the listeners unless we really need to.